text/0000-clarify-api-namespace.md

```diff
@@ -0,0 +1,59 @@
+- Start Date: 2015-08-10
+- RFC PR: (leave this empty)
+- Refract Issue: (leave this empty)
+
+# Summary
+Clarify API description namespace and Iron out issues which rose during API Blueprint serialization to API description namespace.
+
+# Motivation
+The purpose of this document is to clear inconsistencies in the format and make it easier to build tools around it.
+
+# Detailed design
+
+## HREF Variables
+We need to clarify about the value of the `content` in the `hrefVariables` element.
+
+## Data Structure
+We don't need a `Data Structure` because an element which is an `object`, `extend` or `array` can be considered as a data structure element.
+
+*(The current spec is a lot confusing. What is the `Data Structure Element`? Why is `Data Structure` inheriting from it?)*
+
+## Resource
+Since `hrefVariables` is an element, we should move it to be under `content` rather than leave it under `attributes`.
+
+## Transition
+Similarily to the Resource, we need to move `hrefVariables` to be under `content`. `Data Structure` in Resource is under `content`, but it is under `attributes` in Transition. To resolve this inconsistency, we need to move `Data Structure` to be under `content`. Another reason for it is that `Data Structure` is an element.
+
+## Transition Example
+Value of `content` should be changed to `[]` instead of `null`.
+
+## Category
+We need to clarify what `transitions` category is about. If it is being reserved for future use, we need to add reserved keyword.
+
+## Copy
+We need to clarify what the relation between copy elements in `content` and `description` in `attributes` is. Is `description` just an aggregator of all `copy` elements? Why does some examples contain `description` without any `copy` elements and some examples contain `copy` elements but doesn't contain `description`. Since `copy` element is superior, why do we need `description` if we can use `copy` elements everywhere?
```

Comment:
I'm wondering if it may be beneficial to have a video chat about this, because there is actually a lot going on in this.

My first question is, why would you add a copy element to a string? To me, a string is a string. The difficulty here is that we have diverged some from the original structure of the MSON document, though staying with the structure is equally as difficult. Take for instance your example.

```md
# Person (object)
## Description
Copy1 

## Properties
- name

## Description
Copy 2
```

If you didn't have these descriptions, everything would be fine. But once you add them, you also want to retain their structure. This translates to:

* Structure Declaration
    * Copy (Description)
    * Properties
    * Copy (Description)

If you want to retain this order, I'm not sure you can get around this without mimicking this structure. But this really complicates things.

```js
["category", { "class": "element" }, {}, [
  ["copy", {}, {}, "Copy1"],
  ["object", { "id": "Person" }, {}, [
    ["member", {}, {} {
      "key": ["string", {}, {}, "name"],
    }]
  ]],
  ["copy", {}, {}, "Copy2"]
]]
```

Not really proposing this, but my gut feeling is that if you want a solution, you're going to have to go this route or come up with something even more complex.